File: Buoi2/Python/vd3.py
def binary_search(arr, target):
    left, right = 0, len(arr) - 1  # Python dùng chỉ số 0-based
    while left <= right:
        mid = (left + right) // 2  # Tính chỉ số giữa
        if arr[mid] == target:
            return mid
        elif arr[mid] < target:
            left = mid + 1
        else:
            right = mid - 1
    return -1  # Nếu không tìm thấy

arr = [1, 3, 5, 7, 9, 11]
print(binary_search(arr, 7))  # Output: 3

File: Buoi2/C++/vd3.cpp
#include <iostream>
#include <vector>
using namespace std;

int binary_search(const vector<int>& arr, int target) {
    int left = 0, right = arr.size() - 1;  // Chỉ số 0-based
    while (left <= right) {
        int mid = left + (right - left) / 2;  // Tính chỉ số giữa
        if (arr[mid] == target) {
            return mid;
        } else if (arr[mid] < target) {
            left = mid + 1;
        } else {
            right = mid - 1;
        }
    }
    return -1;  // Nếu không tìm thấy
}

int main() {
    vector<int> arr = {1, 3, 5, 7, 9, 11};
    cout << binary_search(arr, 7) << endl;  // Output: 3
    return 0;
}
